src/Microsoft.AspNetCore.Razor.Language/RazorProjectItem.cs

```diff
@@ -18,15 +19,23 @@ public abstract class RazorProjectItem
         public abstract string BasePath { get; }
 
         /// <summary>
-        /// File path relative to <see cref="BasePath"/>.
+        /// File path relative to <see cref="BasePath"/>. This property uses the project path syntax,
+        /// using <c>/</c> as a path separator and does not follow the operating system's file system
+        /// conventions.
         /// </summary>
         public abstract string FilePath { get; }
 
         /// <summary>
-        /// The absolute path to the file, including the file name.
+        /// The absolute physical (file system) path to the file, including the file name.
         /// </summary>
         public abstract string PhysicalPath { get; }
 
+        /// <summary>
+        /// The relative physical (file system) path to the file, including the file name. Relative to the
+        /// physical path of the <see cref="BasePath"/>.
+        /// </summary>
+        public virtual string RelativePhysicalPath => null;
```

Comment:
I'm curious why we are putting file system concepts in `RazorProjectItem` instead of just on `FileSystemRazorProjectItem`. Shouldn't `RazorProjectItem` be generic enough to support items from places other than file system? Or am I misunderstanding the purpose of this abstraction?

Reply:
It already has physical path.

Also, I'm not sure anyone understands the purpose of this abstraction.... If we did it again, I would do it differently. This API has a bunch of MVC-isms built in.... but at the same time, file systems are pretty fundamantal. SO I decided to just embrace it.